modules/core/src/smithy4s/internals/SchemaDescription.scala

```diff
@@ -0,0 +1,164 @@
+    }
+    SchemaDescription.of(value)
+  }
+  override def collection[C[_], A](shapeId: ShapeId, hints: Hints, tag: CollectionTag[C], member: Schema[A]): SchemaDescription[C[A]] = 
+    SchemaDescription.of(tag.name)
+  override def map[K, V](shapeId: ShapeId, hints: Hints, key: Schema[K], value: Schema[V]): SchemaDescription[Map[K,V]] =
+    SchemaDescription.of("Map")
+  
+  override def enumeration[E](shapeId: ShapeId, hints: Hints, values: List[EnumValue[E]], total: E => EnumValue[E]): SchemaDescription[E] =
+    SchemaDescription.of("Enumeration")
+  
+  override def struct[S](shapeId: ShapeId, hints: Hints, fields: Vector[SchemaField[S, _]], make: IndexedSeq[Any] => S): SchemaDescription[S] =
+    SchemaDescription.of("Structure")
+  
+  override def union[U](shapeId: ShapeId, hints: Hints, alternatives: Vector[SchemaAlt[U, _]], dispatch: U => Alt.SchemaAndValue[U, _]): SchemaDescription[U] =
+    SchemaDescription.of("Union")
+  
+  override def biject[A, B](schema: Schema[A], to: A => B, from: B => A): SchemaDescription[B] =
+    SchemaDescription.of(apply(schema))
+  override def surject[A, B](schema: Schema[A], to: Refinement[A,B], from: B => A): SchemaDescription[B] =
+    SchemaDescription.of(apply(schema))
+  override def lazily[A](suspend: Lazy[Schema[A]]): SchemaDescription[A] =
+    suspend.map(s => SchemaDescription.of(apply(s))).value
+}
+
+trait SchemaDescriptionDetailed[A] extends (Set[ShapeId] => (Set[ShapeId], String)) {
+  def mapResult[B](f: String => String): SchemaDescriptionDetailed[B] = { seen =>
+    val (s1, desc) = apply(seen)
+    (s1 ++ seen, f(desc))
+  }
+  def flatMapResult[B](f: String => SchemaDescriptionDetailed[B]): SchemaDescriptionDetailed[B] = { seen =>
+    val (s1, desc) = apply(seen)
+    f(desc)(s1 ++ seen)
+  }
+}
+
+object SchemaDescriptionDetailed extends SchemaVisitor[SchemaDescriptionDetailed] { self =>
+
+  def of[A](shapeId: ShapeId, value: String): SchemaDescriptionDetailed[A] = s => (s + shapeId, value)
+
+  override def primitive[P](shapeId: ShapeId, hints: Hints, tag: Primitive[P]): SchemaDescriptionDetailed[P] = {
+    SchemaDescriptionDetailed.of(
+      shapeId, 
+      SchemaDescription.apply(tag.schema(shapeId))
+    )
+  }
+  override def collection[C[_], A](shapeId: ShapeId, hints: Hints, tag: CollectionTag[C], member: Schema[A]): SchemaDescriptionDetailed[C[A]] = {
+    apply(member).mapResult(s => s"${tag.name}[$s]")
+  }
+  override def map[K, V](shapeId: ShapeId, hints: Hints, key: Schema[K], value: Schema[V]): SchemaDescriptionDetailed[Map[K,V]] = {
+    apply(key).flatMapResult { kDesc =>
+      apply(value).mapResult { vDesc =>
+        s"Map[$kDesc, $vDesc]"
+      }
+    }
+  }
+  override def enumeration[E](shapeId: ShapeId, hints: Hints, values: List[EnumValue[E]], total: E => EnumValue[E]): SchemaDescriptionDetailed[E] = {
+    val vDesc = values.map(e => e.stringValue).mkString(", ")
+    SchemaDescriptionDetailed.of(shapeId, s"Enumeration{ $vDesc }")
```

Comment:
inconsistent whitespace between `Enumeration` and the bracket. For instance, in `Structure`, there's a whitespace between the structure name and the bracket. 

I'm happy for the whitespace to be removed over there.